code/game/machinery/iv_drip.dm

```diff
@@ -172,42 +207,45 @@
 		return PROCESS_KILL
 
 	if(!(get_dist(src, attached) <= 1 && isturf(attached.loc)))
-		to_chat(attached, span_userdanger("The IV drip needle is ripped out of you, leaving an open bleeding wound!"))
-		var/list/arm_zones = shuffle(list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM))
-		var/obj/item/bodypart/chosen_limb = attached.get_bodypart(arm_zones[1]) || attached.get_bodypart(arm_zones[2]) || attached.get_bodypart(BODY_ZONE_CHEST)
-		chosen_limb.receive_damage(3)
-		chosen_limb.force_wound_upwards(/datum/wound/pierce/moderate)
+		if(isliving(attached))
+			var/mob/living/attached_mob = attached
+			to_chat(attached, span_userdanger("The IV drip needle is ripped out of you, leaving an open bleeding wound!"))
+			var/list/arm_zones = shuffle(list(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM))
+			var/obj/item/bodypart/chosen_limb = attached_mob.get_bodypart(arm_zones[1]) || attached_mob.get_bodypart(arm_zones[2]) || attached_mob.get_bodypart(BODY_ZONE_CHEST)
+			chosen_limb.receive_damage(3)
+			chosen_limb.force_wound_upwards(/datum/wound/pierce/moderate)
+		else
+			visible_message(span_warning("[attached] is detached from [src]."))
 		detach_iv()
 		return PROCESS_KILL
 
-	var/datum/reagents/target_reagents = get_reagent_holder()
-	if(target_reagents)
-		// Give blood
+	if(transfer_rate == 0)
+		return
+
+	var/datum/reagents/drip_reagents = get_reagents()
+	if(drip_reagents)
```

Comment:
This seems like an easy early-return
Also uh when will this ever be null?

Reply:
Thanks. It is supposed to be null when the attached container runs out of reagents. Do you think it returns empty list and this check is never negative?

Reply:
When it runs out of reagents there’ll still be a reagent datum, it just wont have any reagents

Reply:
Sorry, I was wrong. This check is there to just check that reagents datum exists. It will be null for the drip without internal storage (regular iv) that was working when someone removed the reagent container.

Reply:
Ah, all good then